seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/FlinkStarter.java

```diff
@@ -0,0 +1,109 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel;
+
+import org.apache.seatunnel.command.FlinkCommandArgs;
+import org.apache.seatunnel.common.config.Common;
+
+import com.beust.jcommander.JCommander;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * The SeaTunnel flink starter. This class is responsible for generate the final flink job execute command.
+ */
+public class FlinkStarter implements Starter {
+
+    private static final String APP_NAME = SeatunnelFlink.class.getName();
+    private static final int USAGE_EXIT_CODE = 234;
+    private static final String APP_JAR_NAME = "seatunnel-core-flink.jar";
+
+    /**
+     * Flink parameters, used by flink job itself. e.g. `-m yarn-cluster`
+     */
+    private final List<String> flinkParams = new ArrayList<>();
+
+    /**
+     * SeaTunnel parameters, used by SeaTunnel application. e.g. `-c config.conf`
+     */
+    private final FlinkCommandArgs flinkCommandArgs;
+
+    /**
+     * SeaTunnel flink job jar.
+     */
+    private final String appJar;
+
+    public FlinkStarter(String[] args) {
+        this.flinkCommandArgs = parseArgs(args);
+        // set the deployment mode, used to get the job jar path.
+        Common.setDeployMode(flinkCommandArgs.getDeployMode().getName());
+        this.appJar = Common.appLibDir().resolve(APP_JAR_NAME).toString();
+    }
+
+    @SuppressWarnings("checkstyle:RegexpSingleline")
+    public static void main(String[] args) {
+        FlinkStarter flinkStarter = new FlinkStarter(args);
+        List<String> command = flinkStarter.buildCommands();
+        String finalFLinkCommand = String.join(" ", command);
+        System.out.println(finalFLinkCommand);
+    }
+
+    /**
+     * Parse seatunnel args.
+     *
+     * @param args args
+     * @return FlinkCommandArgs
+     */
+    private FlinkCommandArgs parseArgs(String[] args) {
+        FlinkCommandArgs flinkCommandArgs = new FlinkCommandArgs();
+        JCommander jCommander = JCommander.newBuilder()
+            .programName("start-seatunnel-flink.sh")
+            .addObject(flinkCommandArgs)
+            .acceptUnknownOptions(true)
+            .args(args)
+            .build();
+        // The args is not belongs to seatunnel, add into flink params
+        flinkParams.addAll(jCommander.getUnknownOptions());
+        if (flinkCommandArgs.isHelp()) {
+            jCommander.usage();
+            System.exit(USAGE_EXIT_CODE);
+        }
+        return flinkCommandArgs;
+    }
+
+    @Override
+    public List<String> buildCommands() {
+        List<String> command = new ArrayList<>();
+        command.add("${FLINK_HOME}/bin/flink run");
+        command.addAll(flinkParams);
+        command.add("-c " + APP_NAME);
+        command.add(appJar);
+        command.add("--config " + flinkCommandArgs.getConfigFile());
```

Comment:
use '+' maybe make command not atomic, split to 2 add step may be better.

Reply:
Ok, modified as you suggested.